### ISSUE TITLE
Limit the number of parallel downloads in the queue to a specific number

### ISSUE THREAD

**Issue by NomarCub** [TeamNewPipe/NewPipe#8744]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

Title says it all. Right now you can only choose between 1 simultaneous download or downloading everything you click on in parallel at the same time. I would like to have the option to limit simultaneous downloads to a specific number.

I saw this mentioned in another issue: https://github.com/TeamNewPipe/NewPipe/issues/4215. I hope this is not a duplicate, I did look around, but there are a ton of issues.

### Why do you want this feature?

It mostly comes down to resource use. 1 at a time doesn't use the device's full potential bandwidth / memory / processing power. I'd like to download faster if my phone can handle it. At the same time, downloading too many videos at the same time can be too resource intensive, make the app unresponsive or even crash, and leave downloads in an unfinished state. You should be able to set a sweet spot of maximum parallel download threads.
